### PR TITLE
add escape sequences in character strings

### PR DESCRIPTION
**Pull request by Kljunas2**
Standard escape sequences (those used in java) and ability to write bytes directly (\xbb).

### CODE REVIEW

File: src/sic/ast/data/DataChr.java [hunk 1]
```diff
@@ -27,7 +27,51 @@ public String toString() {
...
         parser.advance('C');
         parser.advance('\'');
-        data = parser.readUntil('\'').getBytes();
+        StringBuilder buf = new StringBuilder();
+        for (char c = parser.advance(); c != '\''; c = parser.advance()) {
+            if (c == '\\') {
+                c = parser.advance();
+                switch (c) {
+                    case '\'':
+                        c = '\'';
```
Comment (jetomit) [inline src/sic/ast/data/DataChr.java:36]: This looks redundant. :)

**Comment by jetomit** [comment]
I’m inclined to merge this as it seems useful, but it does change the behavior of the assembler a bit (e.g. for `C'\'` and `C'\\'`). @jurem, what do you think?

File: src/sic/ast/data/DataChr.java (hunk 1 above)
Comment (jetomit) [reply src/sic/ast/data/DataChr.java:36]: Line 36, since c already has that value. You could also fall through to the next case.

**Comment by jetomit** [comment]
Unterminated `C'` causes an OutOfMemory exception, not sure what goes wrong.

**Comment by jurem** [comment]
Hi. Thanks! Yes, I think escape seqs support is a useful feature.

Maybe a suggestion not to break the compatibility to any previous assembly code relying on not-escaped strings, a support could be added using two variants:
1) C'...' ... raw strings (as it was originally)
2) C"..." ... strings with escape sequences

You should also probably refactor the actual escaped string parsing (the for loop and big switch) out into a separate function. Maybe add it to the Lexer class, e.g., readEscapedString(char terminator), where terminator is " (double quite as suggested above).

**Comment by Kljunas2** [comment]
Could I then terminate it with zero byte like C strings, or should I leave users the choice to manually add zero byte if desired?

**Comment by jurem** [comment]
I guess user should take care for the zero byte terminator if needed. This is still an assembly, don't be too friendly :)

**Comment by jurem** [comment]
Yes, I guess that's it! Thanks!!!

**Pull request merged**